src/OrchardCore.Modules/OrchardCore.Menu/Controllers/AdminController.cs

```diff
@@ -271,8 +271,9 @@ public async Task<IActionResult> Delete(string menuContentItemId, string menuIte
             return NotFound();
         }
 
-        var menuItems = menuContentAsJson[nameof(MenuItemsListPart)]?[nameof(MenuItemsListPart.MenuItems)] as JsonArray;
-        menuItems?.Remove(menuItem);
+        var menuItems = menuContentAsJson.SelectNode(menuItem.Parent.GetPath()) as JsonArray;
```

Comment:
Count Parent here be null like if you remove the last parent?

Reply:
I already that about that, all the menu items have a parent, let me double-check and delete all the menus in the blog theme

Reply:
I deleted all the menus in the main menu, it works like a charm